File: Golang/Easy/2154_Keep_Multiplying_Found_Values_by_Two/solution.go
package keepmultiplyingfoundvaluesbytwo

func findFinalValue(nums []int, original int) int {
	numMap := make(map[int]int)

	for _, num := range nums {
		numMap[num] += 1
	}

	for true {
		if _, exist := numMap[original]; exist {
			original = original * 2
		} else {
			break
		}
	}

	return original
}

File: Python/Easy/2154_Keep_Multiplying_Found_Values_by_Two/solution.py
from typing import List

class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        numMap = {}
        for num in nums:
            if num not in numMap:
                numMap[num] = 0
            numMap[num] += 1
        
        while original in numMap:
            original = original * 2
        
        return original